azavea/raster-vision
# Pre-release fixes

## Overview

This PR 
- fixes bugs discovered during testing for the 0.30 release
- updates tutorial notebooks
- makes some other minor changes (see commits)

### Checklist

- [ ] Added `needs-backport` label if the change should be back-ported to the previous release

### Notes

Eval comparison with previous release.
```
Comparing
- s3://raster-vision/examples/0.21/output/spacenet-rio-cc
- s3://raster-vision/examples/0.30.0/output/spacenet-rio-cc
Fetching eval directory: s3://raster-vision/examples/0.21/output/spacenet-rio-cc/eval
download: s3://raster-vision/examples/0.21/output/spacenet-rio-cc/eval/validation_scenes/eval.json to ../data/examples/collect/old/spacenet-rio-cc/eval/validation_scenes/eval.json
Fetching eval directory: s3://raster-vision/examples/0.30.0/output/spacenet-rio-cc/eval
download: s3://raster-vision/examples/0.30.0/output/spacenet-rio-cc/eval/validation_scenes/eval.json to ../data/examples/collect/spacenet-rio-cc/eval/validation_scenes/eval.json
Comparing keys and values in eval.json files...
All keys match
426 matching keys found
Ignoring keys that contain ['conf_mat', 'count', 'per_scene']
Float comparison tolerance: 0.01
All values within tolerance
Comparing
- s3://raster-vision/examples/0.21/output/isprs-potsdam-ss
- s3://raster-vision/examples/0.30.0/output/isprs-potsdam-ss
Fetching eval directory: s3://raster-vision/examples/0.21/output/isprs-potsdam-ss/eval
download: s3://raster-vision/examples/0.21/output/isprs-potsdam-ss/eval/validation_scenes/eval.json to ../data/examples/collect/old/isprs-potsdam-ss/eval/validation_scenes/eval.json
Fetching eval directory: s3://raster-vision/examples/0.30.0/output/isprs-potsdam-ss/eval
download: s3://raster-vision/examples/0.30.0/output/isprs-potsdam-ss/eval/validation_scenes/eval.json to ../data/examples/collect/isprs-potsdam-ss/eval/validation_scenes/eval.json
Comparing keys and values in eval.json files...
All keys match
1184 matching keys found
Ignoring keys that contain ['conf_mat', 'count', 'per_scene']
Float comparison tolerance: 0.01
diff: overall.0.metrics.recall: 0.882652 - 0.895635  = -0.012983
diff: overall.0.metrics.sensitivity: 0.882652 - 0.895635  = -0.012983
diff: overall.2.metrics.precision: 0.851947 - 0.864028  = -0.012081
diff: overall.2.metrics.recall: 0.898540 - 0.888205  = 0.010336
diff: overall.2.metrics.sensitivity: 0.898540 - 0.888205  = 0.010336
diff: overall.5.metrics.f1: 0.396004 - 0.432545  = -0.036541
diff: overall.5.metrics.precision: 0.412175 - 0.430002  = -0.017827
diff: overall.5.metrics.recall: 0.381054 - 0.435119  = -0.054065
diff: overall.5.metrics.sensitivity: 0.381054 - 0.435119  = -0.054065
Number of non-matching values: 9
Comparing
- s3://raster-vision/examples/0.21/output/spacenet-vegas-buildings-ss
- s3://raster-vision/examples/0.30.0/output/spacenet-vegas-buildings-ss
Fetching eval directory: s3://raster-vision/examples/0.21/output/spacenet-vegas-buildings-ss/eval
download: s3://raster-vision/examples/0.21/output/spacenet-vegas-buildings-ss/eval/validation_scenes/eval.json to ../data/examples/collect/old/spacenet-vegas-buildings-ss/eval/validation_scenes/eval.json
Fetching eval directory: s3://raster-vision/examples/0.30.0/output/spacenet-vegas-buildings-ss/eval
download: s3://raster-vision/examples/0.30.0/output/spacenet-vegas-buildings-ss/eval/validation_scenes/eval.json to ../data/examples/collect/spacenet-vegas-buildings-ss/eval/validation_scenes/eval.json
Comparing keys and values in eval.json files...
All keys match
83268 matching keys found
Ignoring keys that contain ['conf_mat', 'count', 'per_scene']
Float comparison tolerance: 0.01
All values within tolerance
Comparing
- s3://raster-vision/examples/0.21/output/spacenet-vegas-roads-ss
- s3://raster-vision/examples/0.30.0/output/spacenet-vegas-roads-ss
Fetching eval directory: s3://raster-vision/examples/0.21/output/spacenet-vegas-roads-ss/eval
download: s3://raster-vision/examples/0.21/output/spacenet-vegas-roads-ss/eval/validation_scenes/eval.json to ../data/examples/collect/old/spacenet-vegas-roads-ss/eval/validation_scenes/eval.json
Fetching eval directory: s3://raster-vision/examples/0.30.0/output/spacenet-vegas-roads-ss/eval
download: s3://raster-vision/examples/0.30.0/output/spacenet-vegas-roads-ss/eval/validation_scenes/eval.json to ../data/examples/collect/spacenet-vegas-roads-ss/eval/validation_scenes/eval.json
Comparing keys and values in eval.json files...
All keys match
21276 matching keys found
Ignoring keys that contain ['conf_mat', 'count', 'per_scene']
Float comparison tolerance: 0.01
All values within tolerance
Comparing
- s3://raster-vision/examples/0.21/output/cowc-potsdam-od
- s3://raster-vision/examples/0.30.0/output/cowc-potsdam-od
Fetching eval directory: s3://raster-vision/examples/0.21/output/cowc-potsdam-od/eval
download: s3://raster-vision/examples/0.21/output/cowc-potsdam-od/eval/validation_scenes/eval.json to ../data/examples/collect/old/cowc-potsdam-od/eval/validation_scenes/eval.json
Fetching eval directory: s3://raster-vision/examples/0.30.0/output/cowc-potsdam-od/eval
download: s3://raster-vision/examples/0.30.0/output/cowc-potsdam-od/eval/validation_scenes/eval.json to ../data/examples/collect/cowc-potsdam-od/eval/validation_scenes/eval.json
Comparing keys and values in eval.json files...
All keys match
88 matching keys found
Ignoring keys that contain ['conf_mat', 'count', 'per_scene']
Float comparison tolerance: 0.01
diff: overall.0.false_neg: 27.000000 - 16.000000  = 11.000000
diff: overall.0.false_pos: 75.000000 - 37.000000  = 38.000000
diff: overall.0.metrics.f1: 0.903592 - 0.948594  = -0.045002
diff: overall.0.metrics.precision: 0.864376 - 0.929658  = -0.065282
diff: overall.0.metrics.recall: 0.946535 - 0.968317  = -0.021782
diff: overall.0.metrics.sensitivity: 0.946535 - 0.968317  = -0.021782
diff: overall.0.true_pos: 478.000000 - 489.000000  = -11.000000
diff: overall.1.metrics.f1: 0.903592 - 0.948594  = -0.045002
diff: overall.1.metrics.precision: 0.864376 - 0.929658  = -0.065282
diff: overall.1.metrics.recall: 0.946535 - 0.968317  = -0.021782
diff: overall.1.metrics.sensitivity: 0.946535 - 0.968317  = -0.021782
Number of non-matching values: 11
Comparing
- s3://raster-vision/examples/0.21/output/xview-od
- s3://raster-vision/examples/0.30.0/output/xview-od
Fetching eval directory: s3://raster-vision/examples/0.21/output/xview-od/eval
download: s3://raster-vision/examples/0.21/output/xview-od/eval/validation_scenes/eval.json to ../data/examples/collect/old/xview-od/eval/validation_scenes/eval.json
Fetching eval directory: s3://raster-vision/examples/0.30.0/output/xview-od/eval
download: s3://raster-vision/examples/0.30.0/output/xview-od/eval/validation_scenes/eval.json to ../data/examples/collect/xview-od/eval/validation_scenes/eval.json
Comparing keys and values in eval.json files...
All keys match
308 matching keys found
Ignoring keys that contain ['conf_mat', 'count', 'per_scene']
Float comparison tolerance: 0.01
diff: overall.0.false_neg: 5672.000000 - 5686.000000  = -14.000000
diff: overall.0.false_pos: 7240.000000 - 6423.000000  = 817.000000
diff: overall.0.metrics.f1: 0.641552 - 0.655906  = -0.014354
diff: overall.0.metrics.precision: 0.614791 - 0.642452  = -0.027660
diff: overall.0.true_pos: 11555.000000 - 11541.000000  = 14.000000
diff: overall.1.metrics.f1: 0.641552 - 0.655906  = -0.014354
diff: overall.1.metrics.precision: 0.614791 - 0.642452  = -0.027660
Number of non-matching values: 7
```

## Testing Instructions
Check tutorials in the [docs build](https://raster-vision--2105.org.readthedocs.build/en/2105/).
